packages/xdl/src/Versions.js

```diff
@@ -3,16 +3,35 @@
  */
 
 import _ from 'lodash';
+import path from 'path';
 import semver from 'semver';
 
 import Api from './Api';
 import ApiV2Client from './ApiV2';
+import { Cacher } from './tools/FsCache';
 import ErrorCode from './ErrorCode';
 import XDLError from './XDLError';
 import UserManager from './User';
 
 export async function versionsAsync() {
-  return await Api.versionsAsync();
+  const api = new ApiV2Client();
+  const versionCache = new Cacher(
+    () => api.getAsync('versions/latest'),
+    'versions.json',
+    0,
+    path.join(__dirname, '../caches/versions.json')
```

Comment:
If the format of the result has changed, maybe we should use a different filename here? (E.g. `latest-versions.json` or `versions-v2.json`). This way any data cached by old versions wouldn’t break things.

Also look at the XDL gulpfile which saves this file to `xdl` folder (when publishing), for offline mode. Might have to change the name there as well.

Reply:
The format of the raw response has changed but that's handled by ApiV2Client, so no changes should be necessary here. Tested by running `Versions.versionsAsync()` before and after this change, format of `versions.json` remained the same.

Reply:
👍